corehq/apps/ota/views.py

```diff
@@ -184,7 +184,8 @@ def _add_case_property_queries(domain, case_type, search_es, criteria):
             case_property=key,
         )
         for removal_regex in remove_char_regexs:
-            value = re.sub(removal_regex.regex, '', value)
+            to_remove = re.escape(removal_regex.regex)
```

Comment:
It looks like `regex` attribute on IgnorePatterns is a misnomer, it means a substring to ignore (at least as it's described in [the UI](https://india.commcarehq.org/a/enikshay/settings/project/case_search/) and not a regex. This is something that's configured by users who we assume don't know regex and that input is not treated as regex (for e.g. it's not valdiated if it's a valid regex or not). Should we be treating the input as regex?

```
>>> re.sub("+", '', "+91")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/sravfeyn/HQ/virtualenvs/hqenv/lib/python2.7/re.py", line 155, in sub
    return _compile(pattern, flags).sub(repl, string, count)
  File "/Users/sravfeyn/HQ/virtualenvs/hqenv/lib/python2.7/re.py", line 251, in _compile
    raise error, v # invalid expression
sre_constants.error: nothing to repeat
>>> cut = lambda x, y: re.sub(re.escape(x), '', y)
>>> cut(".exe", "python.exe")
'python'
>>> cut("+", "+91")
'91'
>>> cut("-", "011-91")
'01191'
>>> cut("word", "some word")
'some '
>>> cut("/", "some/word")
'someword'
```